packages/cloud_firestore/cloud_firestore_web/lib/src/interop/firestore.dart

```diff
@@ -109,9 +111,18 @@ class Firestore extends JsObjectWrapper<firestore_interop.FirestoreJsImpl> {
     return LoadBundleTask.getInstance(jsObject.loadBundle(bundle));
   }
 
-  Future<Query> namedQuery(String name) async {
-    firestore_interop.QueryJsImpl query =
+  Future<Query?> namedQuery(String name) async {
```

Comment:
`Query` probably should still be non-nullable

Reply:
even if it can be `null`? There is also a `null` check [just below it](https://github.com/FirebaseExtended/flutterfire/blob/00f0ab6f0c6b29c96e9664e0d5939ba5c7bb033f/packages/cloud_firestore/cloud_firestore_web/lib/src/interop/firestore.dart#L118-L124).

Reply:
But the function cannot return null. The null-check throws an exception

Reply:
this test [indicates](https://github.com/FirebaseExtended/flutterfire/blob/50183df79f73c0840b9330022064c78f462db012/packages/cloud_firestore/cloud_firestore/example/test_driver/load_bundle_e2e.dart#L122) that it does return `null` 😄.

Reply:
I'm wondering if we're talking about the same thing.

It's about `Future<Query?>` into `Future<Query>`

Reply:
Yeah, talking about the same thing. I assume I'm misunderstanding, but if the result of the future can be `null`. Then the type has to be `Future<Query?>`, right? The test also proves it as the `query == null` condition throws the error necessary for the test to pass.

Reply:
Just to be clear, I have also tested on `namedQuery` on web. It will return `null` if the named query does not exist.

Reply:
ok, ignore me, I'm being a dummy.